R/Viewer.R

```diff
@@ -0,0 +1,166 @@
+# given an environment and a host, return the name of an open odbc connection
+find_object <- function(env, host) {
+  objs <- ls(env)
+  for (name in objs) {
+    x <- base::get(name, envir = env)
+    if (inherits(x, "OdbcConnection") &&
+        identical(to_host(x), host) &&
+        dbIsValid(con)) {
+      return(name)
+    }
+  }
+}
+
+# connection-specific actions possible with odbc connections
+odbc_actions <- function() {
+  icons <- system.file(file.path("icons"), package = "odbc")
+  list(
+    Help = list(
+      icon = file.path(icons, "help.png"),
+      callback = function() {
+        utils::browseURL("https://github.com/rstats-db/odbc")
+      }
+    )
+  )
+}
+
+on_connection_opened <- function(con, env, parentCall) {
+  parentCall[[1]] <- as.symbol("dbConnect")
```

Comment:
There is no guarantee that this creates a call that can be re-run in the future. Can I get more details about how you're thinking about the overall design here?